### PR TITLE
mix start

### PR DESCRIPTION
**Pull request by berkserbet**
### Why:
* Mixed customer type implementation.

### What:
* Adds a new mix pipeline, plotting has been added.

### Checklist
- [x] Finished My Changes
- [ ] Automated Unit Tests

### CODE REVIEW

File: generics/artifact.py [hunk 1]
```diff
@@ -84,11 +84,17 @@ def _read_from_s3(self, filename):
...
                 os.rename(temp_filename, f'{base.LOCAL_PATH}/{filename}')
             except botocore.exceptions.ClientError as e:
-                if e.response['Error']['Code'] == "404":
-                    logger.exception(f'File {filename} does not exist in S3 under {base.REMOTE_PATH}')
-                else:
-                    logger.exception(e)
-                raise e
+                try:
+                    temp_filename = str(uuid.uuid4())
+                    s3 = boto3.client('s3')
```
Comment (jongoncalves) [inline generics/artifact.py]: DRY this up. Move the s3 client creation and temp filename assignment up above.

File: pipelines/mix/mix.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+from pipelines.mix.tasks import (
+    get_mixed
+)
+
+
+logger = logging.getLogger('LCTK_APPLICATION_LOGGER')
+
+
+class MixPipeline():
```
Comment (jongoncalves) [inline pipelines/mix/mix.py]: Naming things are hard. Is there a more descriptive name for this pipeline?
Like ResidentialCommercialAnalysisPipeline? <- not literally this, but you get the idea

File: pipelines/mix/tasks/get_mixed.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+    def get_mixed_output(self, mix_chart): 
+        """This function adapt the customer_chart
+        """
+
+        output_df = pd.DataFrame(columns=['target', 'daytype', 'time'] + self.components)
+        
+        for location in self.residential_components.target.unique():
+            
+            initialization = False
```
Comment (jongoncalves) [inline pipelines/mix/tasks/get_mixed.py]: There's examples in the RBSA pipeline on how to bypass the need for this flag. Check it out and update this flow.

File: pipelines/mix/tasks/get_mixed.py [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+                    buildingtype = self.buildingtype_dict[building]
+
+                    if buildingtype == 'RES':
+                        df = self.residential_components.loc[self.residential_components.target == location]
+                    else:
+                        df = self.commercial_components.loc[(self.commercial_components.target == location) & (self.commercial_components.buildingtype == buildingtype)]
+                    
+                    df = df.reset_index() 
+
+                except:
```
Comment (jongoncalves) [inline pipelines/mix/tasks/get_mixed.py]: What error are you expecting here? Don't catch generally. Catch a specific error, like ValueError or TypeError.

File: pipelines/mix/mix.py (hunk 2 above)
Comment (berkserbet) [reply pipelines/mix/mix.py]: changing it to MixedFeederPipeline

File: pipelines/mix/tasks/get_mixed.py (hunk 4 above)
Comment (berkserbet) [reply pipelines/mix/tasks/get_mixed.py]: I have a fix, getting rid of exception.

File: pipelines/mix/tasks/get_mixed.py (hunk 3 above)
Comment (berkserbet) [reply pipelines/mix/tasks/get_mixed.py]: Are you sure, I couldn't find it?

File: pipelines/mix/mix.py (hunk 2 above)
Comment (jongoncalves) [reply pipelines/mix/mix.py]: This is the commit: https://github.com/slacgismo/loadinsight/commit/7dbbdec84b5ea2dfb9a756f0e514ec9cb258fb07

File: pipelines/mix/tasks/get_mixed.py (hunk 3 above)
Comment (jongoncalves) [reply pipelines/mix/tasks/get_mixed.py]: This is the commit: https://github.com/slacgismo/loadinsight/commit/7dbbdec84b5ea2dfb9a756f0e514ec9cb258fb07

**Comment by jongoncalves** [comment]
This branch has been merged into the DEBUG branch. Closing this now.

**Pull request closed without merging**